internal/access/signup.go

```diff
@@ -7,35 +7,64 @@ import (
 	"golang.org/x/crypto/bcrypt"
 
 	"github.com/infrahq/infra/internal"
-	"github.com/infrahq/infra/internal/logging"
 	"github.com/infrahq/infra/internal/server/data"
 	"github.com/infrahq/infra/internal/server/models"
 	"github.com/infrahq/infra/uid"
 )
 
+// SignupEnabled queries the current state of the service and returns whether
+// or not to allow unauthenticated signup. Signup is enabled if and only if
+// the configuration flag 'enableSignup' is set and no identities, providers,
+// or grants have been configured, both currently and previously.
 func SignupEnabled(c *gin.Context) (bool, error) {
 	// no authorization is setup yet
 	db := getDB(c)
 
-	settings, err := data.GetSettings(db)
+	enableSignupVar, ok := c.Get("enableSignup")
+	if !ok {
+		return false, nil
+	}
+
+	enableSignup, ok := enableSignupVar.(bool)
+	if !ok {
+		return false, nil
+	}
```

Comment:
I learned yesterday there were more ways of getting things out of the `gin.Context`. I think this could use:

```suggestion
	enableSignup := c.GetBool("enableSignup")
```

Although avoiding using the context for this might also be nice (as Jeff suggested in another comment).